### PR TITLE
Make search take focus on opening pokedex; add cmd-f shortcut

### PR DESCRIPTION
**Pull request by phoenixeliot**
I added some convenience functionality because I wanted it; I think it makes the website a little bit nicer to use.

This adds some convenience behavior:
- Cmd/Ctrl-F changes page to the Pokedex page
- The search input is focused by default, or if you press cmd-f again on the pokedex page
- The input text is selected by default, so you don't have to delete the contents before typing a new search

Basically, this makes the pokedex search page behave roughly like the browser find-in-page.

### CODE REVIEW

**Comment by wavebeem** [comment]
Hey Phoenix, thanks for working on this. Here's some thoughts about the PR:

> Cmd/Ctrl-F changes page to the Pokedex page

This seems good to me. I think you could remove the OS check and just bind to `Ctrl-F` _and_ `Cmd-F` though. It's simpler and helps out people who are less familiar with whichever OS they happen to be using right now. Also, `/` is a pretty common search shortcut that would be worth binding.

> The search input is focused by default, or if you press cmd-f again on the pokedex page

Fun fact, it used to be focused by default. But this is really annoying on phones/tablets since it forces the keyboard to pop up and cover half the screen. We could get into the game of guessing if the device is being used with a touch keyboard, but it's further complicated by stuff like touch screen laptops...

> The input text is selected by default, so you don't have to delete the contents before typing a new search

This is handy, but I don't think we should do it given the above^

***

FYI, you're welcome to add yourself to the "Special Thanks" part of the Info page.

File: src/App.tsx [hunk 1]
```diff
@@ -31,6 +40,18 @@ export default function App() {
     CoverageType[]
   >([]);
   const [AllPokemon, setAllPokemon] = React.useState<Pokemon[]>([]);
+
+  // Navigate to pokedex page on pressing cmd/ctrl-f
+  useShortcut(
+    (event: KeyboardEvent): void => {
```
Comment (wavebeem) [inline src/App.tsx:46]: Can be simplified to `event => {` because the type signature is inferred based on the definition of `useShortcut`

File: src/Search.tsx [hunk 2]
```diff
@@ -18,6 +19,23 @@ export default function Search({ updateSearch, search }: SearchProps) {
...
   const inputHeight = 36;
+
+  // Select contents of search input when switching to the page
+  React.useEffect(() => {
+    ref?.current?.select();
+  }, [ref]);
+
+  // Select contents of search input when pressing cmd/ctrl-f
+  useShortcut(
+    (event: KeyboardEvent): void => {
```
Comment (wavebeem) [inline src/Search.tsx:30]: Can be simplified to `event => {` because the type signature is inferred based on the definition of `useShortcut`

**Comment by phoenixeliot** [comment]
Thanks for the quick response!
> Fun fact, it used to be focused by default. But this is really annoying on phones/tablets since it forces the keyboard to pop up and cover half the screen. We could get into the game of guessing if the device is being used with a touch keyboard, but it's further complicated by stuff like touch screen laptops...

How do you feel about a "good enough/mostly right" solution here, using something like heuristics based on screen size? Maybe if the first pokemon result would be at all covered by a hypothetical keyboard (that is, fits in ~the top half of the screen), don't autofocus?

In lieu of that, we could make it autofocus only if the user navigated with cmd-f, and not if they use the mouse / tap the screen.

**Comment by wavebeem** [comment]
Yeah, making Cmd-F do (1. Take you to Pokdex; 2. Focus search field; 3. Select all text) sounds perfect to me. Quick shortcut to help folks out, but no chance of bothering folks who don't care.

Maybe you could add another query param, or use some global state in App, so that the Pokedex knew what it should do on load.

**Comment by wavebeem** [comment]
I'm closing this due to the branch being out of date and no recent activity. Feel free to make a new PR if you still want to work on this feature. Thanks!
